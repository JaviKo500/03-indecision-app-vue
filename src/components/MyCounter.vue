<template>
   <section>
      <h3>Counter: {{  counter  }}</h3>
      <h3>Square: {{ squareCounter }}</h3>
      <small>{{ text }}</small>
      <div>
         <!-- <button @click="increment">+1</button>
         <button @click="decrement">-1</button> -->
         <button class="btn" @click="counter++">+1</button>
         <button class="btn" @click="counter--">-1</button>
      </div>
   </section>
</template>

<script lang="ts" setup>
   import { computed, ref } from 'vue';

   interface Props {
      value: number;
      text: string;
   }

   const props = defineProps<Props>();

   // const props = defineProps({
   //    value: {
   //       type: Number,
   //       default: 1,
   //    }
   // });

   console.log('<--------------- JK MyCounter --------------->');
   console.log('props.value', props);
   const counter = ref(props.value );
   const squareCounter =  computed(() => counter.value * counter.value);

   // function increment() {
   //    counter.value++;
   // }

   // function decrement() {
   //    counter.value--;
   // }

</script>
<style scoped >
   .btn {
      @apply p-5 bg-blue-500 rounded hover:bg-blue-700 mr-2;
   }
</style>