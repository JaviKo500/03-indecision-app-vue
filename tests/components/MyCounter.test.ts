import { describe, test } from 'vitest';
import { mount } from '@vue/test-utils';
import MyCounter from '@/components/MyCounter.vue';

describe('MyCounter.test', () => {
   test('should match snapshot', () => {
      const wrapper = mount(MyCounter, {
         props: {
            initialValue: 10,
         },
      });

      console.log('<--------------- JK MyCounter.test --------------->');
      console.log(wrapper.html());
   });
});