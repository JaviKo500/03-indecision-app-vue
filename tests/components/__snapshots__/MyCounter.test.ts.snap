// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MyCounter.test > should match snapshot 1`] = `
"<section data-v-0352012b="" initialvalue="10">
  <h3 data-v-0352012b="">Counter: 5</h3>
  <h3 data-v-0352012b="" data-test-id="square-label">Square: 25</h3><!-- <small>{{ text }}</small> -->
  <div data-v-0352012b="">
    <!-- <button @click="increment">+1</button>
         <button @click="decrement">-1</button> --><button data-v-0352012b="" class="btn">+1</button><button data-v-0352012b="" class="btn">-1</button>
  </div>
</section>"
`;
